### PR TITLE
forgotPassword bug fix

### PR DESCRIPTION
**Pull request by minsunny**
https://nanopay.atlassian.net/browse/NP-4599

nanopay: https://github.com/nanoPayinc/NANOPAY/pull/12843

1. fixed reset password email not being sent out --> findTemplate's group fix in DAOResourceLoader
2. fixed a bug that reset password link in reset password email gets auto log-in 
3. fixed errorPageview not working with error message, updated to use theme logo and ui consistency like SuccessPasswordView
4. added "isTokenValid()" in token service to see if token is expired, used, not existing etc.

2&3&4 updates in requestLogin() controller.js

5. tableWidth in token data not to be cut off

Screen Shot 2021-06-03 at 11 37 06 PM

Screen Shot 2021-06-03 at 11 38 22 PM

Screen Shot 2021-06-03 at 11 39 34 PM

Screen Shot 2021-06-07 at 1 56 28 PM

Screen Shot 2021-06-07 at 1 56 10 PM

Screen Shot 2021-06-07 at 1 55 56 PM

Screen Shot 2021-06-07 at 1 55 30 PM

### CODE REVIEW

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 1]
```diff
@@ -376,7 +376,7 @@ foam.CLASS({
       ps.setString(body);
       ParserContext parserX = new ParserContextImpl();
       parserX.set("sb", sbJoin);
-      parserX.set("x", x);
+      parserX.set("x", getX());
```
Comment (smirnovaka) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: I dont think this is right

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 2]
```diff
@@ -89,8 +89,10 @@ public static EmailTemplate findTemplate(X x, String name, String groupId, Strin
   }
 
   public static EmailTemplate findTemplate(X x, String name) {
-    var groupId = ((Group) x.get("group")).getId();
-    String locale = ((Subject) x.get("subject")).getRealUser().getLanguage().getCode();
+    User user = ((Subject) x.get("subject")).getRealUser();
```
Comment (smirnovaka) [inline src/foam/nanos/notification/email/DAOResourceLoader.java:92]: looks good. wonder how it worked before.

File: src/foam/nanos/notification/email/EmailTemplateEngine.js (hunk 1 above)
Comment (minsunny) [reply src/foam/nanos/notification/email/EmailTemplateEngine.js]: @penzital  updated DAOResourceLoader as we discussed

File: src/foam/nanos/notification/email/DAOResourceLoader.java (hunk 2 above)
Comment (smirnovaka) [reply src/foam/nanos/notification/email/DAOResourceLoader.java:92]: why user context wouldn't have "group"

Comment (minsunny) [reply src/foam/nanos/notification/email/DAOResourceLoader.java:92]: @penzital I think emailTemplateEngine used to find  via emailTemplateDAO not DAOResourceLoader before.

Comment (minsunny) [reply src/foam/nanos/notification/email/DAOResourceLoader.java:92]: talked to @penzital and resolved.

**Review by smirnovaka — APPROVED** [review]
lgtm

**Comment by minsunny** [comment]
this closed and moved https://github.com/kgrgreer/foam3/pull/452/files

**Pull request closed without merging**